poker-pals/src/components/pokerComponents/ChipDistributionCalculator.vue

```diff
@@ -0,0 +1,157 @@
+            <input type="text" id="player4" size="10">
+            <input type="text" id="player5" size="10">
+            <input type="text" id="player6" size="10">
+        </div>
+        <div class="box">
+            <button v-on:click="calculate()">get number</button>
+            <p id="result">0</p>
+        </div>
+    </div>
+</template>
+
+<style scoped>
+
+    .box{
+        width: 923px;
+        margin: auto;
+    }
+    .rank{
+        margin: 0px 0px 0px 53px;
+    }
+
+</style>
+
+<script>
+
+export default {
+    name: "ChipDistributionCalculator",
+    mounted(){
+
+    },
+    methods:{
+        calculate(){
+            
+            let players = [
+                {id: 1, rank: 0x500000, moneyPot: 10, returnPot: 0},
+                {id: 2, rank: 0x900000, moneyPot: 3, returnPot: 0},
+                {id: 3, rank: 0x600000, moneyPot: 3, returnPot: 0},
+                {id: 4, rank: 0x900000, moneyPot: 2, returnPot: 0},
+                {id: 5, rank: 0x700000, moneyPot: 1, returnPot: 0},
+                {id: 6, rank: 0x800000, moneyPot: 4, returnPot: 0}
+            ];
+            
+            let slots = [];
+            slots.push([]);
+            slots.push([]);
+            slots.push([]);
+            slots.push([]);
+            slots.push([]);
+            slots.push([]);
+            
+            players.sort(function(a, b){return b.rank - a.rank});
+            let rankTag = -1;
+            for(let i = 0; i < players.length; i++){
+                if(players[i].rank === rankTag){
+                    // this player matches the previous player in rank
+                    rankTag = players[i].rank;
+                    players[i].rank = players[i-1].rank;
+                }
+                else{
+                    if(i !== 0){
+                        rankTag = players[i].rank;
+                        players[i].rank = players[i-1].rank + 1;
+                    }
+                    else{
+                        rankTag = players[i].rank;
+                        players[i].rank = 1;
+                    }
+                }
+            }
+            for(let i = 0; i < players.length; i++){
+                slots[players[i].rank - 1].push({
+                    id: players[i].id,
+                    rank: players[i].rank,
+                    moneyPot: players[i].moneyPot,
+                    returnPot: 0
+                });
+            }
+            console.log(JSON.stringify(slots));
+
+
+
+            // the chip distribution is determined by a set of rules
+            // the first rule is the pot is represented by six individual chip slots (one for each seat)
+            // the players are sorted into winning ranks, and the first group of winners each show their wagers,
+
+            // the lowest wager starts first by removing that wager value from all other players (include those who didn't get first)
+            for(let i = 0; i < slots.length; i++){
```

Comment:
a lot of the logic in this file can probably be extracted to a function inside this file to make the code more readable